### DISCUSSION TITLE
Quais as melhores distros Linux para desenvolvimento?

### DISCUSSION THREAD

**Discussion by jaocarli** [frontendbr/forum#1744] [Discussão]
Olá meus queridos!

> Primeiramente não esqueçam de lavar as mãos e **não saiam de casa**, somente se for realmente **necessário**!

Bom, vamos lá: estou querendo começar a usar Linux **somente** para programação, deixando o windows somente para jogar. Quais são as distros que vocês utilizam e indicam para quem está começando no mundo do Linux?

Estou pensando em usar ou o Ubuntu mesmo ou o Linux Mint...

deixem as sugestões abaixo aí e vamos trocar uma ideia sobre isso =D

**Comment by bmnascimento** [comment]
Gosto mais do Linux Mint que do Ubuntu por causa da interface Cinnamon que me agrada mais, mas a distro que eu uso mesmo é o Debian, principalmente pelo fato de ser muito estável, depois que vc configura não tem mais dor de cabeça.

**Comment by portothree** [comment]
Uso debian com xfce, bspwm e compton.
Gosto da facilidade de pegar uma versão stable, usar meus [dotfiles](https://github.com/portothree/dotfiles) e ter meu setup pronto em menos de 20min em qualquer maquina.
 
![Current desk screenshot](https://i.imgur.com/ti2LDBF.png)

**Comment by tresvezesheber** [comment]
Utilizo o Fedora desde a versão 29 com o ambiente gráfico padrão (GNOME). Creio ser um ambiente bem tranquilo de se operar, principalmente para iniciantes. Especificamente para o desenvolvimento, posso dizer que nunca tive problema algum em utilizar qualquer tecnologia, linguagem ou ferramentas no Fedora. E essa qualidade estende-se a todo o resto do sistema, resumindo-se em um sistema estável e versátil, para mim.

**Comment by ir4nbala** [comment]
Como você está iniciando no Linux o caminho mais fácil seria as distros
Ubuntu e Linux mint, mas se procura estabilidade tem o Debian.

**Comment by valkervieira** [comment]
O consenso é que Ubuntu e Linux Mint são mais indicados para iniciantes. Mas é sempre bom lembrar que a escolha da distro e o desktop é absolutamente pessoal, e tirando as distros que precisam ser instaladas por linha de comando como o Arch e o Gentoo, é uma questão de testar e encontrar a que você se identifica melhor.

Eu, mesmo quando era iniciante, não consegui me adaptar com o Ubuntu (que é baseado no Debian) e embora o Arch não seja indicado para iniciantes, eu só consegui me adaptar com ele. Então por anos eu instalei o Arch manualmente mas hoje utilizo o Manjaro pq não tenho mais tempo/paciência pra instalar por linha de comando.

O principal motivo pelo qual eu uso distribuições baseadas em Arch é o AUR, um repositório onde todos os aplicativos ficam, basicamente eu não gosto de adicionar repositórios antes de instalar um app como fazia no Ubuntu, no Arch procuro num lugar só.

Outra coisa importante de notar o que é a diferença entre uma distro rolling release vs fixed release. Nas rolling releases o sistema não trabalha com major versions, tipo o Ubuntu que tem o 18, 19, 20, sendo os pares LTS. As rolling releases vão sendo atualizadas e pronto, deu update no sistema ta com a última versão, o que pra mim é bom, mas é comum que venha com alguns bugs, coisa que é mais difícil com um lts do Ubuntu por exemplo (mas não imune)

Quanto aos desktops:
O Gnome mais recente (que vai vir no Ubuntu 20 LTS) está bem performático e bonito, mas naturalmente consome um pouco mais de recursos, a versão que está no Ubuntu 18 também é boa, mas não tanto quanto o 20. O que me incomoda um pouco é que de uma minor version do Gnome pra outra os temas param de funcionar, o que eu acho um porre.
O KDE é o mais customizável, podendo ser um desktop bem leve ou bem pesado dependendo da sua customização. Foi minha escolha por muitos anos junto com o Manjaro. Sentia que entregava mais performance e consumia menos recurso pra entregar uma interface tão ou mais bonita que o Gnome.
O XFCE é leve e ainda assim é bonito, também é bastante customizável e tem ótimos widgets. É a minha escolha atual pra programar e é também o que eu recomendo pq deixando ele básico e bonito dificilmente você vai passar de 600mb de ram em idle.
Cinnamon (acho que é a padrão do Mint) é sem dúvida a mais fácil de migrar pra quem vem do windows. Eu particularmente não gosto do visual e não acho que ela entregue uma performance tão boa pra tão pouco que ela parece oferecer.
Sobre os outros desktops não tenho experiência.

Pontos fora da curva que eu acho interessante comentar são:

Elementary OS - É baseado no Ubuntu e usa uma interface própria (pantheon se não me engano). Ele tenta copiar o MacOS e sempre que lança uma nova versão eu baixo na esperança de finalmente ele parecer um sistema finalizado, o que pra mim nunca acontece. Como eu sou usuário de Mac no dia-a-dia tenho a esperança de ter a mesma experiência o que acaba não se cumprindo. Em geral ele é a única distro que eu tive congelamentos constantes. Vi me fóruns que não acontecia só comigo. Ainda assim eu acho uma boa distro que vale a pena dar uma conferida, caso os problemas de freeze não aconteçam com você, pode ser uma boa.

DeepinOS - É uma distribuição chinesa se não me engano que no passado esteve envolvida em uma polêmica de privaciade (roubo de dados etc) mas que pelo jeito não sofre mais disso. É baseada no Debian (não no Ubuntu) e muita gente gosta, o desktop deles é próprio e eu já usei no Ubuntu e achei MUITO bonito. Mas enjoei depois de 1-2 dias. Ainda assim recomendo dar uma olhada.

Depois que eu tomei a decisão e disciplina de usar o windows única e exclusivamente pra jogar. E trabalhar/navegar no linux sinto que meu computador performa muito melhor já que não instalo praticamente nada no Windows.

Boa sorte e espero que encontre sua distro alma-gêmea como eu encontrei no Manjaro. kkkk

**Comment by jaocarli** [comment]
Muito obrigado pelas dicas meus queridos! Eu estou usando o Mint 19.2 Cinnamon, porém para mim, dá umas congeladas do nada, porém não quero colocar o Ubuntu por não gostar muito do jeitão dele. 

Agora estou na vibe de testar o Debian e o Manjaro, como o @borghettinho e o @portothree indicaram! Estarei baixando os dois e meter bala neles para ver quais escolho. Obrigado a todos! =D

volto logo mais para dizer o que achei das mesmas

**Comment by ThePokyWizard** [comment]
Eu utilizei o Elementary OS por muito tempo, possuía uma interface leve, bonita e apps exclusivos na loja deles que eram muito interessantes.
Como é baseado no Ubuntu vários aplicativos funcionam sem problema nele.
Parei de usar apenas por causa de um problema que ele tinha ao enviar arquivos pelo Slack através do Drag and Drop.

**Comment by felipe-heredia** [comment]
## Sobre as distros.

Sou bem novo no mundo Linux também, chego a instalar diversos em meu pendrive para testes.

### Eu não acho que exista a melhor distro.

A minha forma de utilizar é diferente da sua, e assim varia de pessoa para pessoa. Por tanto, acho que não existe a distro perfeita e sim aquela que te entrega com agilidade aquilo que você deseja fazer.

---

Atualmente, venho fazendo de uma forma bem doida.

1. Removi totalmente o Windows da minha máquina (que por sinal é muito fraca).
2. Antes de remover, já tinha o Linux Mint, modificado e bonitinho da forma que eu desejava.
3. No lugar do Windows, coloquei o Pop!_OS, ao lado do Mint.

No momento, estou utilizando as duas, e gostando bastante do POP, mas em minha máquina ocorrem alguns travamentos sem sentido (visto que tenho uns 3 apps instalados apenas). Então, por medo e para manter o POP um pouco mais oorganizado, mantive ele para fazer apenas Design e testes.

Estou utilizando o POP apenas com o Brave Browser e com o Figma dentro do Brave, assim faço apenas Design de interfaces dentro dele, se for fazer código, vou para o Mint.

Instalei o Atom dentro dele e vou fazendo alguns testes de usabilidade, se quero testar um APP ou programa, testo no PoP, para depois enviar para produção (No Mint).

**Comment by kivervinicius** [comment]
Eu uso no trabalho o ubuntu, nunca precisei mudar, porém minha máquina é um I9 Extreme, em casos de máquinas mais fracas o mint é um dos mais recomendados pela agilidade. 

Por mais que esteja procurando sobre o linux em casa eu estou utilizando o windows com WSL 2 e está fantástico pois possui todas facilidades e pacotes do linux e é muito fácil de mudar de ambiente.

**Comment by ir4nbala** [comment]
Como você está começando, eu recomendo o linux Mint, é o Ubuntu melhorado,
caso esteja procurando por estabilidade ai você pode ir de debian, mas
muitas das configurações serão feitas na mão.

Em qui., 14 de mai. de 2020 às 12:04, Kiver Teixeira <
notifications@github.com> escreveu:

**Comment by ninetails** [comment]
O post do @borghettinho foi fenomenal :tada: 

Também comecei com Ubuntu e Mint, porém fiquei um bom tempo nele. Na faculdade, a distro era Debian e não é muito diferente do Ubuntu. Depois acabei migrando pro Antergos (descontinuado) e indo pro Manjaro pois curti muito Arch.

Gostaria de adicionar um ponto: aprender mesmo é na cara e na coragem instalando direto dual boot (ou se vc tiver uma máquina só pra jogo e outra pra programar, instala direto sem dual boot), mas se não tiver coragem tipo eu no começo, recomendo começar virtualizando. Assim vc pode meio que testar cada distro e interface gráfica, apesar de ser mais emocionante instalar e se frustrar, instalar outra distro/interface e por aí vai, aprendendo...

Uma dica que dou é: já pensa em gerenciar seus _dotfiles_ e subir pro github (tomando, claro, cuidado para não subir credenciais).

**Reply by fininhors** [reply]
O que são esses dotfiles que vc comentou?

**Reply by felipe-heredia** [reply]
@fininhors , _Dotfiles_ são arquivos de configuração, são chamados de dotfiles, pois normalmente ficam em pastas ou são arquivos que iniciam com ponto ".", o ponto serve para serem arquivos escondidos.

Exemplo de dotfiles:
- `.zshrc`;
- `.config/nvim/init.lua`;
- `.gitconfig`;

Tem um exemplo no meu GitHub: https://github.com/felipe-heredia/dotfiles

Apesar de a pasta config não possuir um ponto no início, é criado um link simbólico para a pasta `~/.config` ser a pasta `config` desse repositório.

**Comment by leandrocrs** [comment]
Realmente, não existe uma distro melhor para desenvolvimento ou etc.

Mas como usuário GNU/Linux desde 2008 e desenvolvedor frontend, depois de ter experimentado os principais desktop environments, as principais distros (debian-based, rpm-based e arch-linux), cheguei a um denominador comum para iniciantes: **alguma distro baseada em Ubuntu LTS** (Long Time Supported) com **GNOME ou XFCE** inicialmente.

Vai experimentando outros DEs dentro de VMs até achar um que você consiga se identificar. O meu preferido é o KDE, por ser facilmente personalizável do jeito que quero, e por achar ele bem bonito.

Se for muito iniciante e quiser focar em código, vai de Linux Mint ou Ubuntu LTS (18.04 ou 20.04) que não tem erro. Suporte da comunidade é excelente e já topei com apps que só possuíam versão DEB e não RPM, por isso não recomendo Fedora ou openSUSE para quem está começando.

Se quiser aprender sobre GNU/Linux e sistemas operacionais, não deixe de experimentar o Arch-Linux. Instalar e personalizar seu Arch é um divisor de águas na vida de um usuário GNU/Linux, é sério! É como tunar o primeiro carro,

Aproveito e deixo meu setup atual:
![Kubuntu 20.04 + Latte Dock](https://user-images.githubusercontent.com/1456009/82357839-0db45b80-99dc-11ea-8fc7-7674791004bc.png)

**Reply by Gabriel-Moreira-Valverde** [reply]
está area de trabalho é no arch linux e se eu quiser deixar igual o seu eu baixo o gnome ?

**Reply by g4o4ox** [reply]
não, no caso você precisaria instalar o KDE Plasma e fazer algumas personalizações

**Comment by nazarepiedady** [comment]
O sensato é não perder muito tempo com a escolha de sistemas operativos baseados no núcleo Linux, porque já existe um por qual todos deveriam começar e que oferece um suporte muito maior a controladores de sistemas e dispositivos (**drivers**) e o nome é **Ubuntu**.

Quando estiveres familiarizado com a realidade dos sistemas operativos baseados no Linux, estarás pronto para migrar para qualquer outro sistema que despertar o teu interesse.

**Comment by leomaurodesenv** [comment]
Sou apaixonado pela simplicidade do Ubuntu. #change my mind

**Comment by qwirt1zi** [comment]
> Olá meus queridos!
> 
> > Primeiramente não esqueçam de lavar as mãos e **não saiam de casa**, somente se for realmente **necessário**!
> 
> Bom, vamos lá: estou querendo começar a usar Linux **somente** para programação, deixando o windows somente para jogar. Quais são as distros que vocês utilizam e indicam para quem está começando no mundo do Linux?
> 
> Estou pensando em usar ou o Ubuntu mesmo ou o Linux Mint...
> 
> deixem as sugestões abaixo aí e vamos trocar uma ideia sobre isso =D

Olha, uma distro muito boa para você usar é aprender bastante é o Debian puro, ele é bem leve tem um repositório muito eficiente, além de ser muito seguro é muito bom para quem está programando também, como interfere gráfica te recomendo usar o gnome, ou até mesmo uma mais leve, ou uma de sua preferência.
